### PR TITLE
[Google.com_Subdomains] Simplify and reactivate ruleset

### PR DESCRIPTION
**Pull request by Bisaloo**

### CODE REVIEW

File: src/chrome/content/rules/Google.com_Subdomains.xml [hunk 1]
```diff
@@ -86,7 +87,6 @@ Non-2xx HTTP code: http://suggestqueries.google.com/ (200) => https://suggestque
 	<target host="hostedtalkgadget.google.com" />
 	<target host="investor.google.com" />
 	<target host="ipv4.google.com" />
-	<target host="ipv6.google.com" />
```
Comment (cschanaj) [inline src/chrome/content/rules/Google.com_Subdomains.xml]: @Hainish does our Travis support IPv6 currently? Related travis-ci/travis-ci/issues/6082 and EFForg/https-everywhere/pull/11127

cc @koops76 @Bisaloo

Comment (koops76) [inline src/chrome/content/rules/Google.com_Subdomains.xml]: @cschanaj We have `miredo` running while testing so I guess yes.

File: src/chrome/content/rules/Google.com_Subdomains.xml [hunk 2]
```diff
@@ -122,19 +121,7 @@ Non-2xx HTTP code: http://suggestqueries.google.com/ (200) => https://suggestque
...
-			to="https://$1.corp.google.com/" />
-		<test url="http://login.corp.google.com/" />
-		<test url="http://ideas.corp.google.com/" />
-		<test url="http://discovery.corp.google.com/" />
-
-	<!-- subdomain to path redirects -->
-	<rule from="^http://(earth|trends)\.google\.com/"
-		 to="https://www.google.com/$1" />
-		<test url="http://earth.google.com/" />
-		<test url="http://trends.google.com/" />
```
Comment (J0WI) [inline src/chrome/content/rules/Google.com_Subdomains.xml]: Could you re-add them as targets?

File: src/chrome/content/rules/Google.com_Subdomains.xml (hunk 1 above)
Comment (J0WI) [inline src/chrome/content/rules/Google.com_Subdomains.xml]: Yes, please keep this target

**Pull request merged**